### DISCUSSION TITLE
issue: update method in useFieldArray hook clears the value  of the item

### DISCUSSION THREAD

**Discussion by adamhemzal** [react-hook-form/react-hook-form#9720] [Q&A] [answered]
### Version Number

7.41.3

### Codesandbox/Expo snack

https://codesandbox.io/s/field-array-update-problem-3kdbpi

### Steps to reproduce

Small form where you could be able to append and remove answers for a question. You could be able to `update` the answer to mark that the answer is correct.

**How to reproduce the bug?**
1. Open Codesandbox
2. (optional) Open `console.log` where you can see the defaultValues for `answers` when the component is loaded
3. Fill the `question name` input and fill the `answer` input
4. Click on the `Correct` button to update the `correct` value from `false` to `true`
5. **`update` does not hold the value of the input field. The `answer` field is cleared and new `id` is generated. (probably due to unmouting and remounting)**

**Important notes:**
- It might not be a bug, I might do something wrong.
- When I'm using the `setValue` it WORKS but the UI is not updated. I'd like to update the UI (better UX) and keep the values of my inputs.

I would be happy for any help.

Thank you.

### Expected behaviour

1. Fill the `question name` input and fill the `answer` input
2. Click on the `Correct` button to update the `correct` value from `false` to `true` in the array of objects
3. **`update` holds the value of the input field. The `correct` value is changed and the rest of the object stays the same and the UI is updated to see changes for the user.**

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
> update does not hold the value of the input field. The answer field is cleared and new id is generated. (probably due to unmouting and remounting)

update will update key and mount and unmount the inputs.

> When I'm using the setValue it WORKS but the UI is not updated. I'd like to update the UI (better UX) and keep the values of my inputs.

What do you mean? Perhaps you want to subscribe to the field value to react to the changes.

**Comment by adamhemzal** [comment]
> What do you mean? Perhaps you want to subscribe to the field value to react to the changes.

Thank you for quick reply. I want to re-render my UI (change the button name from "Correct" to "Uncorrect") when I click on the button and change the value inside of my array of objects `[{value: "", correct: false}, ...]`  (`fields`). 

I'll try to work on it a little bit more. Hope I find a solution soon. 

Btw.. thanks for the react-hook-form. It's an amazing library

**Comment by leapful** [comment, marked as answer]
@adamhemzal 

The `field` object from `useFieldArray` hook only contains default value and key for the field array item, it does not include the current  value for the field (Reference: https://react-hook-form.com/api/usefieldarray).

Please use `getValues` method from `useForm` hook to get updated value.

Working demo: https://codesandbox.io/s/field-array-update-problem-forked-t5idw3

**Reply by adamhemzal** [reply]
@leapful Thanks a lot! This is the solution. Happy new year 2023
